File: src/main/scala/simple/B5.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;

/**
 * @Author: gin
 * @Date: Created in 下午2:15 19-7-11
 * @Description:
 */
public class B5 {
    public static void main(String[] args) {
        int a[] = new int[10];  // 所求排列组合
        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            list.add(i);
        }
        final int FAC[] = {1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880};   // 阶乘
        int x = 1000000-1;
        int flag=0;
        for (int i = 0; i < 10; i++) {
            int r = x % FAC[10 - 1 - i];
            int t = x / FAC[10 - 1 - i];

            a[i]=list.get(t);
//            System.out.println(t);
//            x=x-t*FAC[10 - 1 - i];
            x=r;
//            System.out.println(x);
            list.remove(t);
//            System.out.println(list);
        }
        for (int i = 0; i < 10; i++) {
            System.out.print(a[i]);
        }
    }
}

File: src/main/scala/Interview/B5.scala
package Interview

import java.util
import java.util.ArrayList

import scala.collection.mutable.ArrayBuffer

/**
  * @Author: chenjinxin
  * @Date: Created in 下午4:59 19-9-29
  * @Description:
  */
object B5 extends App {

  var a = new Array[Char](13) // 所求排列组合
  var i = -1
  var arr = ArrayBuffer.fill(13)({
    i += 1
    ('a' + i).toChar
  })
  //  val FAC = Array(1, 1, 2, 6, 24, 120, 720, 5040, 40320, 362880) // 阶乘
  i = 1
  var j = 1
  val FAC: Array[Long] = Array.concat(Array(1), Array.fill(12)({
    j *= i
    i += 1
    j
  })
  )
  FAC.foreach(println(_))
    var x = 1000000 - 1
    i = 0
    while (i < 13) {
      val r = x % FAC(13 - 1 - i)
      val t = x / FAC(13 - 1 - i)
      a(i) = arr(t.toInt)
      x = r.toInt
      arr -= (a(i))
      i += 1
    }

    a.foreach(print(_))

}
